project/profiles/capacity4more/modules/c4m/wysiwyg/c4m_wysiwyg_ckeditor/c4m_wysiwyg_ckeditor.module

```diff
@@ -104,3 +104,125 @@ function c4m_wysiwyg_ckeditor_media_wysiwyg_format_form_prepare_alter(
       'c4m_wysiwyg_ckeditor'
     ) . '/c4m_wysiwyg_ckeditor.js';
 }
+
+/**
+ * Define all external domains that use cookies.
+ *
+ * @return array
+ *   A list with all domains that contain cookies.
+ */
+function _c4m_wysiwyg_ckeditor_ckeditor_iframes_with_cookies() {
+  return [
+    'youtube',
+    'archive',
+    'dailymotion',
+    'google',
+    'slideshare',
+    'vimeo',
+    'ustream',
+    'prezi',
+    'issuu',
+    'scribd',
+    'tubechop',
+    'twitter',
+    'twitframe',
+    'mapbox',
+    'soundcloud',
+    'audioboom',
+    'mixcloud',
+    'mindmeister',
+  ];
+}
+
+/**
+ * Implements hook_filter_info().
+ */
+function c4m_wysiwyg_ckeditor_filter_info() {
+  $filters['c4m_wysiwyg_ckeditor_iframe_cck'] = array(
+    'title' => t('Iframe Cookie Consent Filter'),
+    'description' => t('Checks iframes and adds the EU Cookie Consent kit to it if applicable'),
+    'process callback' => '_c4m_wysiwyg_ckeditor_iframe_cck_filter_process',
+    'tips callback' => '_c4m_wysiwyg_ckeditor_iframe_cck_filter_tips',
+  );
+
+  return $filters;
+}
+
+/**
+ * Returns tips for defined filter.
+ *
+ * @param string $filter
+ *   Filter for which the tips are added.
+ * @param string $format
+ *   Format of the filter.
+ * @param bool $long
+ *   Decide the type of the tips.
+ *
+ * @return string
+ *   Text that describes ckeditor filter.
+ */
+function _c4m_wysiwyg_ckeditor_iframe_cck_filter_tips(
+  $filter,
+  $format,
+  $long = FALSE
+) {
+  return t('Checks the source input for iframes of a specific service and adds 
+  the EU Cookie Consent kit to it if applicable.');
+}
+
+/**
+ * Prepares markup for external iframes.
+ */
+function _c4m_wysiwyg_ckeditor_iframe_cck_filter_process(
+  $text,
+  $filter,
+  $format,
+  $langcode,
+  $cache,
+  $cache_id
+) {
+  // Check if there is an iframe in the text.
+  if (!preg_match('/<\s*iframe /mis', $text)) {
+    return;
```

Comment:
Please `return $string;` instead of void, otherwise all body is replaced with "".